### PR TITLE
set up cypress and initial code coverage configuration

### PR DESCRIPTION
**Pull request by behr-davide**
🌴🌴🌴🌴
---
If you want to see the code coverage in action:
- `$ cd frontend && npm cypress`
- Run the login.spec.js fixture (currently it just visits the login page)
- `$ open coverage/lcov-report/index.htm`

### CODE REVIEW

File: frontend/cypress/integration/login.spec.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+describe('login tests', () => {
+    it('should log in with a valid username & password', () => {
```
Comment (domkinger) [inline frontend/cypress/integration/login.spec.js:2]: Why are we committing this test with no real function to it?

Comment (behr-davide) [reply frontend/cypress/integration/login.spec.js:2]: I was using it to verify that the code coverage was working. I'll push a commit to remove that spec file.

Comment (domkinger) [reply frontend/cypress/integration/login.spec.js]: we can clean up the commented code and leave the test for now or we can add a real login test... lets not remove the whole thing though.

File: frontend/cypress/integration/login.spec.js [hunk 2]
```diff
@@ -1,8 +0,0 @@
-describe('login tests', () => {
```
Comment (domkinger) [inline frontend/cypress/integration/login.spec.js]: I think we should add this back and just remove the comments, or flesh it out. What do you think?

**Pull request merged**